Source/core/ProcessInfo.cpp

```diff
@@ -61,60 +61,63 @@ namespace Core {
         return (result);
     }
 #elif defined(__APPLE__)
-    void ProcessName(const uint32_t pid, TCHAR buffer[], const uint32_t maxLength)
+    static string ProcessName(const uint32_t pid)
     {
-        proc_pidpath(pid, buffer, maxLength);
+        char pathbuf[PROC_PIDPATHINFO_MAXSIZE];
+
+        proc_name(pid, pathbuf, sizeof(pathbuf));
+        return pathbuf;
+
     }
     static string ExecutableName(HANDLE handle)
     {
         char pathbuf[PROC_PIDPATHINFO_MAXSIZE];
 
-        ProcessName(handle, pathbuf, sizeof(pathbuf));
+        proc_pidpath(handle, pathbuf, sizeof(pathbuf));
         return pathbuf;
     }
 
 #else
-    void ProcessName(const uint32_t pid, TCHAR buffer[], const uint32_t maxLength)
+    static string ProcessName(const uint32_t pid)
     {
-        ASSERT(maxLength > 0);
-
-        if (maxLength > 0) {
-            char procpath[48];
-            snprintf(procpath, sizeof(procpath), "/proc/%u/exe", pid);
-            ssize_t length = readlink(procpath, buffer, maxLength - 1);
-
-            if (length > 0) {
-                buffer[length] = '\0';
-            } else {
-                int fd;
+        char name[PATH_MAX];
 
-                snprintf(procpath, sizeof(procpath), "/proc/%u/comm", pid);
+        int fd;
 
-                if ((fd = open(procpath, O_RDONLY)) > 0) {
-                    ssize_t size;
-                    if ((size = read(fd, buffer, maxLength - 1)) > 0) {
-                        if (buffer[size - 1] == '\n') {
-                            buffer[size - 1] = '\0';
-                        } else {
-                            buffer[size] = '\0';
-                        }
-                    } else {
-                        buffer[0] = '\0';
-                    }
+        std::stringstream ss;
```

Comment:
I know I did something similar later in this file ("pathToSmaps"), but on second thought I think we could replace this with string creation, as it seems to be more performant. 

https://quick-bench.com/q/SI-XzZyfd7YimTzp6dbdEQG__xo

Let me know what do you think and if you agree, could you also change stringstream I created?